crates/rustls/src/tls_stream.rs

```diff
@@ -78,7 +79,12 @@ pub fn tls_connect(
     dns_name: String,
     client_config: Arc<ClientConfig>,
 ) -> (
-    Pin<Box<dyn Future<Output = Result<TlsStream<TokioTlsClientStream>, io::Error>> + Send>>,
+    Pin<
```

Comment:
Maybe we should consider making this an `async fn`?

Reply:
I will try it out tomorrow.

Reply:
In tls_stream.rs, there is `async fn connect_tls()' already. 
For `fn tls_connect`, as it returns a tuple, and the returned BufStreamHandle is not a future. It does not work to make this `fn tls_connect` async up to the chain. IMHO.

One thing I can modify is to return impl Future instead of Pin<Box<dyn Futre<>>, though I am not sure if it's desired.

Reply:
I can clean that up later. I missed that that was a Tuple.